parsl/dataflow/dflow.py

```diff
@@ -714,6 +714,22 @@ def atexit_cleanup(self):
         if not self.cleanup_called:
             self.cleanup()
 
+    def cleanup_wait_for_all_tasks(self):
```

Comment:
consider naming this `wait_for_all_tasks` since it's not doing any particular cleanup